source/Fleet.h

```diff
@@ -45,6 +45,14 @@ class System;
 // AI personality, and set of friendly and hostile "hail" messages, and the ship
 // names are chosen based on a given random "phrase" generator.
 class Fleet {
+public:
+	// Struct to hold an instance of a fleet.
+	struct FleetHolder {
+		std::string name;
+		std::shared_ptr<Ship> flagship;
+		std::set<std::shared_ptr<Ship>> ships;
```

Comment:
This should be a vector or list, so you don't lose the information about the ordering of the ships in the fleet. Given that the number is small, a vector is probably the fastest.

Also, `shared_ptr` comparisons use the pointer value, not the object, so using an ordered set is pointless. If you do want to discard the information about the ordering, then you should use an unordered_set instead. That'll also get rid of the tree management that is usually more cost than it is worth when the number of objects is small.

Reply:
done